build/dsc/resource.rb

```diff
@@ -186,8 +186,25 @@ def ps_module
         end
 
         module_dir = path_array[0..(path_array.count - (index + 2))].join('/')
-        module_manifest_path = "#{module_dir}/#{module_name}.psd1"
-        raise "module manifest #{module_manifest_path} not found" unless File.exists?(module_manifest_path)
+
+        # A psd1 manifest is assumed to be the same name as the DSC Resource
+        # However on case sensitive OS, the file name could be a different
+        # case than the DSC Resource name, i.e SqlServerDsc vs SQLServerDsc.ps1,
+        # which would cause File.exists? to fail to find the file even though
+        # it exists.
+        #
+        # Dir.glob is used here to find any psd1 manifest in the module_dir
+        # folder. This is a safe assumption because by convention there can
+        # only be one manifest file for a DSC Resource because it follows
+        # PowerShell Module manifest conventions. However there could be cases
+        # where there are other psd1 files for different purposes, so we
+        # filter for ones that match the module_name exactly, lowercased
+        module_manifest_path = Dir.glob("#{module_dir}/*.psd1")
+                                  .select{ |fn| (File.basename(fn, ".psd1")).casecmp(module_name) == 0 }
```

Comment:
Since I updated my original comment, not sure if you saw that `.select{}.first` can be simplified to `.find` with the same predicate?

Reply:
Didn't see that either.